server/src/main/java/org/elasticsearch/action/admin/indices/rollover/RolloverRequest.java

```diff
@@ -153,10 +153,6 @@ public ActionRequestValidationException validate() {
             );
         }
 
-        if (failureStoreOptions.includeFailureIndices() && lazy) {
-            validationException = addValidationError("lazily rolling over a failure store is currently not supported", validationException);
```

Comment:
I'm having trouble envisioning a scenario where that would protect against unwanted/erroneous behavior. Could you sketch a scenario perhaps?

Reply:
This is protecting us in the case when a cluster is being upgraded and not all nodes support this feature yet.

Reply:
Sorry I should have elaborated. I understand that node features are useful for such cases in general. My question was more about this specific case. If we would add this check to the `TransportRolloverAction#masterOperation`, it means that this check would only be executed on master nodes that have this feature already. Rollover/lazy rollover is only dependent on the master node, right? So, I don't see in what scenario we would be executing this check, but we can't/shouldn't perform the rollover.

Reply:
One scenario I could think about is that this master has been upgraded but you do not know if the other masters have. 

The problem with these things is that it's hard to argue what would happen if a new feature is used in a partially upgraded cluster, because it's very hard to think all the possible scenarios and failovers. By not protecting this you have two problems:
1. The user things the feature is properly supported, you accepted the request so it should be fine.
2. This request changes changes the cluster state by setting a pre-existing flag to `true` but it also has a later side-effect, the actual rollover which might or might not be implemented at that node either.

Does this make sense?

Reply:
Ah yeah that definitely makes sense. Thanks for the explanation!